aiogram/types/message.py

```diff
@@ -1582,19 +1582,15 @@ def link(self, text, as_html=True) -> str:
         :param parse_mode:
         :return:
         """
-        kwargs = {"chat_id": chat_id, "reply_markup": reply_markup or self.reply_markup}
+        kwargs = {"chat_id": chat_id, "reply_markup": reply_markup or self.reply_markup, "parse_mode": parse_mode or ParseMode.HTML}
         text = self.text or self.caption
 
         if disable_notification is not None:
             kwargs["disable_notification"] = disable_notification
         if reply_to_message_id is not None:
             kwargs["reply_to_message_id"] = reply_to_message_id
-        if parse_mode is not None:
-            kwargs["parse_mode"] = parse_mode
-            if parse_mode == 'html':
-                text = self.html_text if (self.text or self.caption) else None
-            if parse_mode == 'markdown':
-                text = self.md_text if (self.text or self.caption) else None
+        if not kwargs.get("reply_markup"):
```

Comment:
I think it is not necessary, since we already [have a check](https://github.com/aiogram/aiogram/blob/dev-2.x/aiogram/utils/payload.py#L26) for None.